util/syspolicy/syspolicy_test.go

```diff
@@ -67,13 +74,246 @@ func TestGetString(t *testing.T) {
 			if value != tt.value {
 				t.Fatalf("got value %v instead of expected value %v", value, tt.value)
 			}
-			if err != tt.err {
+			if err != tt.err && tt.err != ErrNoSuchKey {
 				t.Fatalf("got error %v instead of expected error %v", err, tt.err)
 			}
 		})
 	}
 }
 
+func TestGetUint64(t *testing.T) {
+	tests := []testHandler{
+		{
+			t:    t,
+			name: "read existing value",
+			key:  KeyExpirationNoticeTime,
+			u64:  1,
+			err:  nil,
+		},
+		{
+			t:    t,
+			name: "read non-existing value",
+			key:  LogSCMInteractions,
+			u64:  0,
+			err:  ErrNoSuchKey,
+		},
+		{
+			t:    t,
+			name: "reading value returns other error",
+			key:  FlushDNSOnSessionUnlock,
+			u64:  0,
+			err:  errors.New("blah"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler = &tt
+			value, err := GetUint64(tt.key, 0)
+			if err != nil && tt.err == ErrNoSuchKey {
+				t.Fatalf("got %v error instead of handling ErrNoSuchKey", err)
+			}
+			if value != tt.u64 {
+				t.Fatalf("got value %v instead of expected value %v", value, tt.u64)
+			}
+			if err != tt.err && tt.err != ErrNoSuchKey {
+				t.Fatalf("got error %v instead of expected error %v", err, tt.err)
+			}
+		})
+	}
+}
+
+func TestGetPreferenceOption(t *testing.T) {
+	tests := []testHandler{
+		{
+			t:                t,
+			name:             "always by policy",
+			key:              EnableIncomingConnections,
+			value:            "always",
+			preferenceOption: alwaysByPolicy,
+			err:              nil,
+		},
+		{
+			t:                t,
+			name:             "never by policy",
+			key:              EnableIncomingConnections,
+			value:            "never",
+			preferenceOption: neverByPolicy,
+			err:              nil,
+		},
+		{
+			t:                t,
+			name:             "user default",
+			key:              EnableIncomingConnections,
+			value:            "user-decides",
+			preferenceOption: showChoiceByPolicy,
+			err:              nil,
+		},
+		{
+			t:                t,
+			name:             "read non-existing value",
+			key:              EnableIncomingConnections,
+			preferenceOption: showChoiceByPolicy,
+			err:              ErrNoSuchKey,
+		},
+		{
+			t:                t,
+			name:             "other error is returned",
+			key:              EnableIncomingConnections,
+			value:            "user-decides",
+			preferenceOption: showChoiceByPolicy,
+			err:              errors.New("blah"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler = &tt
```

Comment:
Watch out, this won't do what you expect. It makes handler set to the pointer of the iterator variable, which means the next time through this loop it'll have the value of a new test automatically even if you only `handler = &tt` once. This causes problems most often if you run parallel subtests, which can lead to accidentally only running the last subtest a bunch of times, in parallel.